src/components/Timer/Timer.tsx

```diff
@@ -1,68 +1,68 @@
-import React, { FC, useCallback, useEffect, useState } from 'react';
+import React, { useCallback, useEffect } from 'react';
 import dayjs from 'dayjs';
 import { TimerDefaultType } from '../../Mocs/TimerMock';
 
 type PropsTimerType = {
   timeDefault: TimerDefaultType;
+  setIsRunning: (isRunning: boolean) => void;
+  isRunning: boolean;
+  currentTime: TimerDefaultType;
+  setCurrentTime: (time: TimerDefaultType) => void;
 };
 
-export const Timer: FC<PropsTimerType> = ({ timeDefault }) => {
-  const [time, setTime] = useState(timeDefault);
-  const [timer, setTimer] = useState(false);
-  const [disabledButton, setDisableButton] = useState(false);
+export const Timer = React.memo(
+  ({
+    timeDefault,
+    setIsRunning,
+    isRunning,
+    currentTime,
+    setCurrentTime,
+  }: PropsTimerType) => {
+    const minutesForEndDate = +timeDefault.minutes;
+    const secondsForEndDate = +timeDefault.seconds;
 
-  const minutesForEndDate = +timeDefault.minutes;
-  const secondsForEndDate = +timeDefault.seconds;
+    let endDate = dayjs().add(minutesForEndDate, 'minute');
+    endDate = endDate.add(secondsForEndDate, 'seconds');
 
-  let endDate = dayjs().add(minutesForEndDate, 'minute');
-  endDate = endDate.add(secondsForEndDate, 'seconds');
+    const calculateTimeLeft = useCallback(() => {
+      const now = dayjs();
+      const seconds = endDate.diff(now, 'seconds') % 60;
+      const minutes = endDate.diff(now, 'minutes') % 60;
 
-  const calculateTimeLeft = useCallback(() => {
-    const now = dayjs();
-    const seconds = endDate.diff(now, 'seconds') % 60;
-    const minutes = endDate.diff(now, 'minutes') % 60;
+      if (seconds <= 0 && minutes <= 0) {
+        setIsRunning(!isRunning);
+        setCurrentTime({ seconds: '00', minutes: '00' });
+      } else {
+        const numberWithZero = (number: number) => {
```

Comment:
please transfer this function to util. We don't need create it for every [endDate, isRunning] change

Reply:
I gave the task to Pavel https://github.com/gitkoshelew/CC.ui/issues/77